### DISCUSSION TITLE
custom migrations expect `./?/init.lua` in the path setting

### DISCUSSION THREAD

**Discussion by Tieske** [Kong/kong#7340] [Ideas and feature requests]
If you create a custom plugin in a non standard location and set `lua_package_path = /usr/local/custom/?.lua`, then migrations fail. Since the core file is `init.lua`, and is required as `kong.plugins.<name>.migrations`.

So the migrations expect the `init.lua` to be part of the path; `lua_package_path = /usr/local/custom/?.lua;/usr/local/custom/?/init.lua`.

The `init.lua` is a common Lua idiom, but should not be required for custom plugins.

**Comment by dndx** [comment]
Merged in https://github.com/Kong/kong/pull/6993.
